pkg/knode-manager/adapters/k8s/pvc.go

```diff
@@ -0,0 +1,102 @@
+package k8sadapter
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/kosmos.io/kosmos/pkg/knode-manager/utils"
+)
+
+type PVCAdapter struct {
+	master kubernetes.Interface
+	client kubernetes.Interface
+
+	updatedPVC chan *corev1.PersistentVolumeClaim
+
+	stopCh <-chan struct{}
+}
+
+func NewPVCAdapter(ctx context.Context, ac *AdapterConfig) (*PVCAdapter, error) {
+	adapter := &PVCAdapter{
+		master: ac.Master,
+		client: ac.Client,
+		stopCh: ctx.Done(),
+	}
+
+	cInformer := kubeinformers.NewSharedInformerFactory(adapter.client, 0).Core().V1().PersistentVolumeClaims()
+	_, _ = cInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		//AddFunc:    adapter.addPVC,
+		UpdateFunc: adapter.updatePVC,
+		//DeleteFunc: adapter.deletePVC,
+	})
+
+	return adapter, nil
+}
+
+func (p *PVCAdapter) updatePVC(oldObj, newObj interface{}) {
+	newPV := newObj.(*corev1.PersistentVolumeClaim)
+	if newPV.Annotations == nil || newPV.Annotations["global"] != "true" {
+		return
+	}
+
+	if newPV.Namespace == metav1.NamespaceSystem || newPV.Name == "kubernetes" {
+		key, err := cache.MetaNamespaceKeyFunc(newObj)
+		if err != nil {
+			runtime.HandleError(err)
+			return
+		}
+		klog.Info("kosmos update pv, enqueue: %v", key)
+		//p.downstreamQueue.Add(key)
```

Comment:
updatePVC seems like doing nothing

Reply:
code missing, it should be p.updatedPVC <- newPVC, I changed it